flexmeasures/cli/data_add.py

```diff
@@ -1118,6 +1121,200 @@ def add_schedule_for_storage(
+    type=click.Path(),
+    help="Path to the output file of the results of the report."
+    "Use the `.csv` suffix to save the results as Comma Separated Values and `.xlsx` to export them as Excel sheets.",
+)
+@click.option(
+    "--timezone",
+    "timezone_optional",
+    help="timezone as string, e.g. 'UTC' or 'Europe/Amsterdam' (defaults to FLEXMEASURES_TIMEZONE config setting)",
+)
+@click.option(
+    "--last-hour",
+    "last_hour",
+    is_flag=True,
+    help="Use the last known 1 hour period for the scope of the report.",
+)
+@click.option(
+    "--last-day",
+    "last_day",
+    is_flag=True,
+    help="Use the last known 1 day period for the scope of the report.",
+)
+@click.option(
+    "--last-week",
+    "last_week",
+    is_flag=True,
+    help="Use the last known 7 days period for the scope of the report.",
+)
+@click.option(
+    "--last-month",
+    "last_month",
+    is_flag=True,
+    help="Use the last known 1 month period for the scope of the report.",
+)
+@click.option(
+    "--last-year",
+    "last_year",
+    is_flag=True,
+    help="Use the last known 1 year period for the scope of the report.",
+)
+@click.option(
+    "--save-to-database",
+    "save_to_database",
+    is_flag=True,
+    help="Whether to save the report to the database.",
+)
+def add_report(
+    reporter_class: str,
+    sensor: Sensor,
+    reporter_config_file: Path,
+    start: Optional[datetime] = None,
+    end: Optional[datetime] = None,
+    resolution: Optional[timedelta] = None,
+    output_file: Optional[Path] = None,
+    save_to_database: bool = False,
+    last_hour: bool = False,
+    last_day: bool = False,
+    last_week: bool = False,
+    last_month: bool = False,
+    last_year: bool = False,
+    timezone_optional: Optional[str] = None,
+):
+    """
+    Create a new report using the Reporter class and save the results
+    to the database or export them as csv or excel file.
+    """
+
+    check_timezone(timezone_optional)
+
+    tz = pytz.timezone(
+        zone=timezone_optional
+        if timezone_optional
+        else app.config.get("FLEXMEASURES_TIMEZONE", "UTC")
+    )
+
+    start, end = get_timerange_from_flags(
+        start,
+        end,
+        tz,
+        last_hour=last_hour,
+        last_day=last_day,
+        last_week=last_week,
+        last_month=last_month,
+        last_year=last_year,
+    )
+
+    click.echo(f"Report scope:\n\tstart={start}\n\tend={end}")
+
+    click.echo(
+        f"Looking for the Reporter {reporter_class} among all the registered reporters.",
+    )
+
+    ReporterClass = app.reporters.get(reporter_class)
+
+    if ReporterClass is None:
+        click.secho(
+            f"Reporter class `{reporter_class}` not available.",
+            **MsgStyle.ERROR,
+        )
+        raise click.Abort()
+
+    click.secho(f"Reporter {reporter_class} found", **MsgStyle.SUCCESS)
+
+    with open(reporter_config_file, "r") as f:
+        reporter_config_raw = json.load(f)
```

Comment:
I'm not sure about creating a new filed because I think it's too niche.  It would only makes sense for reading (and not writing).

However, I think a related improvement would be to use the click param File, which reads the content and then use `json.loads(file_content)` to convert the content to a dictionary.